hrms/payroll/doctype/salary_slip/salary_slip.py

```diff
@@ -618,21 +698,55 @@ def add_earning_for_hourly_wages(self, doc, salary_component, amount):
 		if not row_exists:
 			wages_row = {
 				"salary_component": salary_component,
-				"abbr": frappe.db.get_value("Salary Component", salary_component, "salary_component_abbr"),
+				"abbr": frappe.db.get_value(
+					"Salary Component", salary_component, "salary_component_abbr", cache=True
+				),
 				"amount": self.hour_rate * self.total_working_hours,
 				"default_amount": 0.0,
 				"additional_amount": 0.0,
 			}
 			doc.append("earnings", wages_row)
 
+	def set_salary_structure_assignement(self):
+		start_date = getdate(self.start_date)
+		date_to_validate = self.joining_date if self.joining_date > start_date else start_date
+		self.salary_structure_assignment = frappe.get_value(
```

Comment:
```suggestion
		self._salary_structure_assignment = frappe.get_value(
```
Let's use a leading underscore for defining this attribute to differentiate it with actual fields in salary slip doctype for readability?
https://dbader.org/blog/meaning-of-underscores-in-python